src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -385,6 +385,12 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, AfterViewIni
     ) {
       event.stopPropagation();
     }
+
+    // Special case if click Escape, if input is empty, close the dropdown, if not, empty out the search field
+    if (event.keyCode === ESCAPE && this.value) {
```

Comment:
this should be enabled only when a corresponding optional input is set, in order to keep the current behavior, e.g.
```
@Input() enableClearOnEscapePressed = false
```

Reply:
Okay i will change it :)